jupyter_ascending/handlers/server_extension.py

```diff
@@ -0,0 +1,131 @@
+from pathlib import Path
+from typing import Any
+from typing import Dict
+from typing import Optional
+
+from jsonrpcclient import request
+from jsonrpcserver import dispatch
+from jsonrpcserver import method
+from notebook.base.handlers import IPythonHandler  # type: ignore
+from notebook.utils import url_path_join  # type: ignore
+
+from jupyter_ascending._environment import SYNC_EXTENSION
+from jupyter_ascending.errors import UnableToFindNotebookException
+from jupyter_ascending.functional import get_matching_tail_tokens
+from jupyter_ascending.logger import J_LOGGER
+
+_REGISTERED_SERVERS: Dict[str, int] = {}
+
+
+def _clear_registered_servers():
+    global _REGISTERED_SERVERS
+
+    _REGISTERED_SERVERS = {}
+
+
+class JupyterAscendingHandler(IPythonHandler):
+    # def get(self):
+    #     print("got get request!")
+    #     self.finish('Hello, world!')
+
+    def post(self) -> None:
+        """We receive commands as HTTP POST requests.
+
+        TODO: might want to convert this to async in the future
+        (tornado and jsonrpcserver support asyncio)"""
+        request = self.request.body.decode()
+        response = dispatch(request)
+        J_LOGGER.info("Got Response:\n\t\t{}", response)
+        if response.wanted:
+            self.write(str(response))
+
+    def check_xsrf_cookie(self):
+        """Disable XSRF cookie checking on this request type"""
```

Comment:
Erm, is this safe and reasonable, hard for me to reason about. I think so? But worth thinking about just a bit more in case we're missing something (thought I doubt there's really a way around this?)

Reply:
Yeah, this is a good point to raise. Not the XSRF specifically, but more generally we don't have any auth/login requirement for the jupyter-ascending data channel. So in theory if someone is running a jupyter server that's open to the internet, an attacker could pretty easily check if the server has jupyter-ascending installed and then send up+execute remote code on that machine.

But.. figuring out how to get our client lib + editor integrations to support all the various jupyter auth methods sounds like a huge pain. If the token auth uses static tokens maybe that one wouldn't be too bad.. but I hate to require users to use a specific auth method just to use our plugin.

Reply:
For now maybe I should just make a very visible note on the Readme to be aware of how this works?

Reply:
I think jupyter ports are already super insecure, so yeah, I'm ok with a note saying "really, do not make this accessible on the internet, use an ssh tunnel"